scripts/buildDocs.ts

```diff
@@ -111,24 +112,43 @@ const formatterDocumentation: IDocumentation = {
  */
 function buildDocumentation(documentation: IDocumentation) {
     // Create each module's documentation file.
-    const paths = glob.sync(documentation.globPattern);
-    const metadataJson = paths.map((path: string) => {
-        return buildSingleModuleDocumentation(documentation, path);
-    });
+    const modulePaths = glob.sync(documentation.globPattern);
+    const metadataJson = modulePaths.map((modulePath: string) =>
+        buildSingleModuleDocumentation(documentation, modulePath));
+
+    // Delete outdated directories
+    const rulesDirs = metadataJson.map((metadata: any) => metadata[documentation.nameMetadataKey]);
+    deleteOutdatedDocumentation(documentation.subDirectory, rulesDirs);
 
     // Create a data file with details of every module.
     buildDocumentationDataFile(documentation, metadataJson);
 }
 
 /**
+ * Deletes directories which are outdated
+ * @param directory Path from which outdated subdirectories have to be checked and removed
+ * @param rulesDirs The names of the current and new rules documentation directories
+ */
+function deleteOutdatedDocumentation(directory: string, rulesDirs: string[]) {
+    // find if the thing at particular location is a directory
+    const isDirectory = (source: string) => fs.lstatSync(source).isDirectory();
+    // get all subdirectories in source directory
+    const getDirectories = (source: string) => fs.readdirSync(source).filter((name) => isDirectory(path.join(source, name)));
+
+    const subDirs = getDirectories(directory);
+    const outdatedDirs = subDirs.filter((dir) => rulesDirs.indexOf(dir) < 0);
+    outdatedDirs.forEach((outdatedDir) => rimraf(path.join(directory, outdatedDir), () => {})); // tslint:disable-line:no-empty
```

Comment:
This script uses sync IO everwhere else. Is there a `rimraf.sync`?

Reply:
The options to read all subdirectories within current directory in NodeJS are discussed on StackOverflow at https://stackoverflow.com/questions/18112204

I noticed that rimraf is currently used in tslint clean script, so reused it https://github.com/palantir/tslint/blob/ca42fe978ccebcfba8eed9d27774312f7af8eddd/package.json#L21-L22

Did you mean `rimraf.async`?
I did see a question on StackOverflow which mentions rimraf uses sync I/O https://stackoverflow.com/questions/28282880

Reply:
`rimraf(f, [opts], callback)` is the async API, hence the `callback` parameter.
`rimraf.sync(f)` is the synchronous API you should probably prefer here